store/src/androidTest/kotlin/com/dropbox/external/store5/ShareableLruCacheAndroidTests.kt

```diff
@@ -0,0 +1,230 @@
+class ShareableLruCacheAndroidTests {
+    private val testScope = TestScope()
+    private lateinit var memoryLruCache: ShareableLruCache
+
+    @Before
+    fun before() {
+        memoryLruCache = ShareableLruCache(10)
+    }
+
+    private fun headPointer() = memoryLruCache.head
+    private fun tailPointer() = memoryLruCache.tail
+    private fun head() = memoryLruCache.head.next as ShareableLruCache.Node<Note>
+    private fun tail() = memoryLruCache.tail.prev as ShareableLruCache.Node<Note>
+
+    @Test
+    fun writeAndRead() {
+        testScope.launch {
+            memoryLruCache.write(FakeNotes.One.key, FakeNotes.One.note)
+        }
+        testScope.advanceUntilIdle()
+
+
+        var headPointer = headPointer()
+        var tailPointer = tailPointer()
+        var head = head()
+        var tail = tail()
+
+        assertEquals(ShareableLruCache.headPointer, headPointer)
+        assertEquals(ShareableLruCache.tailPointer, tailPointer)
+        assertEquals(FakeNotes.One.note, head.value)
+        assertEquals(FakeNotes.One.note, tail.value)
+        assertEquals(1, memoryLruCache.cache.size)
+
+        testScope.launch {
+            memoryLruCache.write(FakeNotes.Two.key, FakeNotes.Two.note)
+        }
+        testScope.advanceUntilIdle()
+
+
+
+        headPointer = headPointer()
+        tailPointer = tailPointer()
+        head = head()
+        tail = tail()
+
+        assertEquals(ShareableLruCache.headPointer, headPointer)
+        assertEquals(ShareableLruCache.tailPointer, tailPointer)
+        assertEquals(FakeNotes.Two.note, head.value)
+        assertEquals(FakeNotes.One.note, tail.value)
+        assertEquals(2, memoryLruCache.cache.size)
+
+        val result = memoryLruCache.read<Note>(FakeNotes.One.key)
+
+        testScope.runTest {
+            assertEquals(FakeNotes.One.note, result.last())
+        }
+
+        headPointer = headPointer()
+        tailPointer = tailPointer()
+        head = head()
+        tail = tail()
+
+        assertEquals(ShareableLruCache.headPointer, headPointer)
+        assertEquals(ShareableLruCache.tailPointer, tailPointer)
+        assertEquals(FakeNotes.One.note, head.value)
+        assertEquals(FakeNotes.Two.note, tail.value)
+        assertEquals(2, memoryLruCache.cache.size)
+    }
+
+    @Test
+    fun write10() {
+
+        testScope.launch {
+            FakeNotes.listN(10).forEach {
+                memoryLruCache.write(it.key, it.note)
+            }
+        }
+
+        testScope.advanceUntilIdle()
+
+        val headPointer = headPointer()
+        val tailPointer = tailPointer()
+        val head = head()
+        val tail = tail()
+
+        assertEquals(ShareableLruCache.headPointer, headPointer)
+        assertEquals(ShareableLruCache.tailPointer, tailPointer)
+        assertEquals(FakeNotes.Ten.note, head.value)
+        assertEquals(FakeNotes.One.note, tail.value)
+        assertEquals(10, memoryLruCache.cache.size)
+    }
+
+
+    @Test
+    fun write11() {
```

Comment:
what is difference between this test and previous one?

Reply:
Cache was configured with size of 10

Reply:
Renamed